corehq/pillows/mappings/case_mapping.py

```diff
@@ -31,7 +31,7 @@
                                         'referenced_id': {'type': 'string'},
                                         'referenced_type': {'type': 'string'}
                                     },
-                                    'type': 'object'},
+                                    'type': 'nested'},
```

Comment:
This won't be reindexed because the `CASE_INDEX` did not change. @benrudolph this should get included with your PR